File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
      int curr = 0;
      List<Boolean> output = new ArrayList<>();
      for(int i = 0;i < nums.length; i++){
        curr = (curr*2+nums[i]) % 5;
        if (curr == 0)
        output.add(true);
        else
        output.add(false);
      } 
      return output;
    }
}

File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.cs
public class Solution {
    public IList<bool> PrefixesDivBy5(int[] nums) {
        int curr = 0;
        List<bool> output = new List<bool>();
        foreach(int bit in nums){
            curr = (curr * 2 +bit) % 5;
            if (curr == 0)
            output.Add(true);
            else
            output.Add(false);
        }
        return output;
    }
}